pallets/rmrk-core/src/functions.rs

```diff
@@ -238,122 +263,193 @@ impl<T: Config> Nft<T::AccountId, StringLimitOf<T>> for Pallet<T> {
+		ensure!(sender == root_owner, Error::<T>::NoPermission);
+		// Get NFT info
 		let mut sending_nft =
 			NFTs::<T>::get(collection_id, nft_id).ok_or(Error::<T>::NoAvailableNftId)?;
-		ensure!(&sending_nft.rootowner == &sender, Error::<T>::NoPermission);
-
-		match new_owner.clone() {
-			AccountIdOrCollectionNftTuple::AccountId(account_id) => {
-				// Remove previous parental relationship
-				if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
-					sending_nft.owner
-				{
-					if let Some(mut kids) = Children::<T>::take(cid, nid) {
-						kids.retain(|&kid| kid != (collection_id, nft_id));
-						Children::<T>::insert(cid, nid, kids);
-					}
-				}
-				sending_nft.rootowner = account_id.clone();
-			},
+
+		// Prepare transfer
+		let new_owner_account = match new_owner.clone() {
+			AccountIdOrCollectionNftTuple::AccountId(id) => id,
 			AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) => {
-				let recipient_nft = NFTs::<T>::get(cid, nid).ok_or(Error::<T>::NoAvailableNftId)?;
-				// Check if sending NFT is already a child of recipient NFT
+				// Check if NFT target exists
+				ensure!(NFTs::<T>::contains_key(cid, nid), Error::<T>::NoAvailableNftId);
+				// Check if sending to self
+				ensure!((collection_id, nft_id) != (cid, nid), Error::<T>::CannotSendToDescendentOrSelf);
+				// Check if collection_id & nft_id are descendent of cid & nid
 				ensure!(
 					!Pallet::<T>::is_x_descendent_of_y(cid, nid, collection_id, nft_id),
-					Error::<T>::CannotSendToDescendent
+					Error::<T>::CannotSendToDescendentOrSelf
 				);
-
-				// Remove parent if exists: first we only care if the owner is a non-AccountId)
-				if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
-					sending_nft.owner
-				{
-					// second we only care if the parent has children (it should)
-					if let Some(mut kids) = Children::<T>::take(cid, nid) {
-						// third we only "retain" the other children
-						kids.retain(|&kid| kid != (collection_id, nft_id));
-						Children::<T>::insert(cid, nid, kids);
-					}
-				}
-				if sending_nft.rootowner != recipient_nft.rootowner {
-					// sending_nft.rootowner = recipient_nft.rootowner
-					sending_nft.rootowner = recipient_nft.rootowner.clone();
-
-					Pallet::<T>::recursive_update_rootowner(
-						collection_id,
-						nft_id,
-						recipient_nft.rootowner,
-						max_recursions,
-					)?;
-				}
-				match Children::<T>::take(cid, nid) {
-					None => Children::<T>::insert(cid, nid, vec![(collection_id, nft_id)]),
-					Some(mut kids) => {
-						kids.push((collection_id, nft_id));
-						Children::<T>::insert(cid, nid, kids);
-					},
-				}
+				// Convert to virtual account
+				Pallet::<T>::nft_to_account_id::<T::AccountId>(cid, nid)
 			},
 		};
-		sending_nft.owner = new_owner.clone();
 
+		sending_nft.owner = new_owner.clone();
 		NFTs::<T>::insert(collection_id, nft_id, sending_nft);
 
-		Ok((collection_id, nft_id))
+		Ok(new_owner_account)
```

Comment:
Do we take care of the children index to reflect the change?

Reply:
Oh, I see the send operation takes care of that. However I argue it's better to do it here because we need to ensue a call can do all the bookkeeping stuff by default, unless we point it out clearly

Reply:
Moved to `nft_send` in PR #52